tf2/test/simple_tf2_core.cpp

```diff
@@ -34,31 +34,22 @@
 #include "tf2/exceptions.h"
 #include "tf2/time.h"
 
-typedef std::chrono::system_clock::time_point TimePoint;
-
-void seed_rand()
-{
-  //Seed random number generator with current microseond count
-  srand((unsigned int)std::chrono::system_clock::now().time_since_epoch().count());
-};
-
-void generate_rand_vectors(double scale, uint64_t runs, std::vector<double>& xvalues, std::vector<double>& yvalues, std::vector<double>&zvalues)
-{
-  seed_rand();
-  for ( uint64_t i = 0; i < runs ; i++ )
-  {
-    xvalues[i] = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
-    yvalues[i] = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
-    zvalues[i] = 1.0 * ((double) rand() - (double)RAND_MAX /2.0) /(double)RAND_MAX;
-  }
-}
-
-
-
 TEST(tf2, setTransformFail)
 {
   tf2::BufferCore tfc;
   geometry_msgs::msg::TransformStamped st;
+  st.header.frame_id = "foo";
```

Comment:
I think this test was relying on 'bad' values.

Here's the output of an upstream job (http://build.ros.org/job/Ipr__geometry2__ubuntu_trusty_amd64/83/consoleFull search for setTransformFail):

```
16:02:56 [----------] 3 tests from tf2
16:02:56 [ RUN      ] tf2.setTransformFail
16:02:56 Error:   TF_SELF_TRANSFORM: Ignoring transform from authority "authority1" with frame_id and child_frame_id  "" because they are the same
16:02:56          at line 220 in /tmp/catkin_workspace/src/geometry2/tf2/src/buffer_core.cpp
16:02:56 Error:   TF_NO_CHILD_FRAME_ID: Ignoring transform from authority "authority1" because child_frame_id not set 
16:02:56          at line 226 in /tmp/catkin_workspace/src/geometry2/tf2/src/buffer_core.cpp
16:02:56 Error:   TF_NO_FRAME_ID: Ignoring transform with child_frame_id ""  from authority "authority1" because frame_id not set
16:02:56          at line 232 in /tmp/catkin_workspace/src/geometry2/tf2/src/buffer_core.cpp
16:02:56 Error:   TF_DENORMALIZED_QUATERNION: Ignoring transform for child_frame_id "" from authority "authority1" because of an invalid quaternion in the transform (0.000000 0.000000 0.000000 0.000000)
16:02:56          at line 256 in /tmp/catkin_workspace/src/geometry2/tf2/src/buffer_core.cpp
```

But the output of the CI for this job:
```
16:26:36 3: [ RUN      ] tf2.setTransformFail
16:26:36 3: Error:   TF_SELF_TRANSFORM: Ignoring transform from authority "authority1" with frame_id and child_frame_id  "foo" because they are the same
16:26:36 3:          at line 235 in /Users/osrf/jenkins/workspace/ci_osx/ws/src/ros2/geometry2/tf2/src/buffer_core.cpp
16:26:36 3: [       OK ] tf2.setTransformFail (0 ms)
```